### PR TITLE
fix(input): the clear event cannot be invoked when using the clear-ic…

### PR DESCRIPTION
**Pull request by YHoney7**
…on slot

## Checklist

List of tasks you have already done and plan to do.

- [ ] Fix linting errors
- [ ] Tests have been added / updated (or snapshots)

## Change information

Describe your modifications here.

## Issues

The issues you want to close, formatted as close #1.

## Related Links

Links related to this pr.

### CODE REVIEW

**Comment by haoziqaq** [comment]
ok.
please complete unit test, docs, types.

**Comment by YHoney7** [comment]
好的

File: packages/varlet-ui/src/input/example/locale/en-US.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export default {
   disabled: 'Disabled',
   readonly: 'Readonly',
   clearable: 'Clearable',
+  clearIconSlot: 'use the clear icon slot',
```
Comment (chouchouji) [inline packages/varlet-ui/src/input/example/locale/en-US.ts:11]: Use，首字母大写

Comment (YHoney7) [reply packages/varlet-ui/src/input/example/locale/en-US.ts:11]: 好的

File: packages/varlet-ui/types/input.d.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export class Input extends VarComponent {
   $slots: {
     'prepend-icon'(): VNode[]
     'append-icon'(): VNode[]
-    'clear-icon'(): VNode[]
+    'clear-icon'(clear: function): VNode[]
```
Comment (haoziqaq) [inline packages/varlet-ui/types/input.d.ts:50]: function -> (e: Event) => void

Comment (YHoney7) [reply packages/varlet-ui/types/input.d.ts:50]: 收到

File: packages/varlet-ui/src/input/docs/en-US.md [hunk 3]
```diff
@@ -188,7 +188,7 @@ const value10 = ref('')
 | --- | --- | --- |
 | `prepend-icon` | Prepend Icon | `-` |
 | `append-icon` | Append Icon | `-` |
-| `clear-icon` | Clear Icon | `-` |
+| `clear-icon` | Clear Icon | `clear: function` clear function (can be used to clear input box contents when using slots) |
```
[the remark this answers is not in the document]
Comment (YHoney7) [reply packages/varlet-ui/src/input/docs/en-US.md:191]: 收到

File: packages/varlet-ui/src/input/example/index.vue [hunk 4]
```diff
@@ -36,6 +38,11 @@ onThemeChange()
     <var-input :placeholder="t('readonly')" readonly v-model="standardValue3" />
     <var-input :placeholder="t('disabled')" disabled v-model="standardValue4" />
     <var-input :placeholder="t('clearable')" clearable v-model="standardValue5" />
+    <var-input clearable :placeholder="t('clearIconSlot')" v-model="standardValue11">
+      <template #clear-icon="{ clear }">
+        <var-icon name="error" @click="clear" />
+      </template>
+    </var-input>
```
Comment (haoziqaq) [inline packages/varlet-ui/src/input/example/index.vue:45]: example 添加了案例之后，对应的文档也需要提供案例

Comment (YHoney7) [reply packages/varlet-ui/src/input/example/index.vue:45]: 收到

**Review by chouchouji — APPROVED** [review]

File: packages/varlet-ui/src/input/docs/en-US.md [hunk 5]
```diff
@@ -20,14 +20,20 @@ const value7 = ref('')
...
     <var-input placeholder="Please enter text" v-model="value" />
     <var-input placeholder="Readonly" readonly v-model="value2" />
     <var-input placeholder="Disabled" disabled v-model="value3" />
-    <var-input placeholder="Clearable" clearable v-model="value4" />
+    <var-input placeholder="Clearable" clearable v-model="value11" />
+    <var-input clearable placeholder="Use the clear icon slot" v-model="value11">
+      <template #clear-icon="{ clear }">
+        <var-icon name="error" @click="clear" />
+      </template>
+    </var-input>
```
Comment (haoziqaq) [inline packages/varlet-ui/src/input/docs/en-US.md:36]: 注意一下文档的可读性，placeholder 放第一个，value 0-11 的顺序重新排列一下。

Comment (YHoney7) [reply packages/varlet-ui/src/input/docs/en-US.md:36]: 好的

**Comment by haoziqaq** [comment]
Select 组件应该也需要修改

**Comment by YHoney7** [comment]
需要再开个PR吗单独修改select, 还是在当前的PR修改呢

**Comment by chouchouji** [comment]
> Select 组件应该也需要修改

耗子哥，再开个pr吧

**Pull request merged**